internal/cqarrow/arrow.go

```diff
@@ -0,0 +1,150 @@
+package cqarrow
+
+import (
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/goccy/go-json"
+)
+
+const (
+	MetadataPrimaryKey     = "cq:extension:primary_key"
+	MetadataPrimaryKeyTrue = "true"
+)
+
+func CQColumnToArrowField(col *schema.Column) arrow.Field {
+	var typ arrow.DataType
+	metadata := make(map[string]string)
+
+	switch col.Type {
+	case schema.TypeBool:
+		typ = arrow.FixedWidthTypes.Boolean
+	case schema.TypeInt:
+		typ = arrow.PrimitiveTypes.Int64
+	case schema.TypeFloat:
+		typ = arrow.PrimitiveTypes.Float64
+	case schema.TypeUUID:
+		typ = NewUUIDType()
+	case schema.TypeString:
+		typ = arrow.BinaryTypes.String
+	case schema.TypeByteArray:
+		typ = arrow.BinaryTypes.Binary
+	case schema.TypeStringArray:
+		typ = arrow.ListOf(arrow.BinaryTypes.String)
+	case schema.TypeIntArray:
+		typ = arrow.ListOf(arrow.PrimitiveTypes.Int64)
+	case schema.TypeTimestamp:
+		typ = arrow.FixedWidthTypes.Timestamp_us
```

Comment:
Maybe add a comment that `us` means micros seconds (and not US timezone)

Reply:
arrow type is named badly, should've been `arrow.FixedWidthTypes.TimestampMicros`